packages/frontend/src/pages/insight-page/components/insight-viewer/components/insight-change-history/change-history-view.tsx

```diff
@@ -14,29 +14,48 @@
  * limitations under the License.
  */
 
-import { Box, Flex, Heading, HStack, Icon, Text, Tooltip, Stack, VStack } from '@chakra-ui/react';
+import { Box, Flex, Heading, HStack, Icon, Text, Tooltip, Stack, VStack, Button } from '@chakra-ui/react';
 import { DateTime } from 'luxon';
+import { useState } from 'react';
 import { useSelector } from 'react-redux';
 
 import { ExternalLink } from '../../../../../../components/external-link/external-link';
 import { TextWithIcon } from '../../../../../../components/text-with-icon/text-with-icon';
 import { UserTag } from '../../../../../../components/user-tag/user-tag';
-import type { InsightChangeEdge } from '../../../../../../models/generated/graphql';
+import type { Insight, InsightChange, InsightChangeEdge } from '../../../../../../models/generated/graphql';
 import { formatDateIntl, formatRelativeIntl } from '../../../../../../shared/date-utils';
 import { iconFactory } from '../../../../../../shared/icon-factory';
 import type { RootState } from '../../../../../../store/store';
 
 export interface ChangeHistoryProps {
   changeEdge: InsightChangeEdge;
   insightFullName: string;
+  enableRollBack: boolean;
+  onRollBackChange: Promise<Insight>;
+  fetchingRollBack: boolean;
 }
-export const ChangeHistoryView = ({ changeEdge, insightFullName, ...props }) => {
-  const change = changeEdge.node;
+export const ChangeHistoryView = ({
+  changeEdge,
+  insightFullName,
+  enableRollBack,
+  onRollBackChange,
+  fetchingRollBack,
+  ...props
+}) => {
+  const change: InsightChange = changeEdge.node;
 
   const { appSettings } = useSelector((state: RootState) => state.app);
 
+  const [clicked, setClicked] = useState(false);
+
+  const onClick = async () => {
+    setClicked(true);
+    await onRollBackChange(change.oid);
+    setClicked(false);
+  };
+
   return (
-    <>
+    <Stack key={change.abbreviatedOid} {...props}>
```

Comment:
You shouldn't need a key here, there's no loop in this component.  Doesn't hurt, but can be safely removed.